armi/reactor/components/component.py

```diff
@@ -232,11 +232,27 @@ def __init__(
         self.temperatureInC = Thot
         self.material = None
         self.setProperties(material)
+        self.tInputWarning(Tinput)
         self.applyMaterialMassFracsToNumberDensities()  # not necessary when duplicating...
         self.setType(name)
         self.p.mergeWith = mergeWith
         self.p.customIsotopicsName = isotopics
 
+    def tInputWarning(self, Tinput):
```

Comment:
I think we should move this to the unit tests.

Validating the input data during run time makes our code slower. And we can't do ALL of the validation in the world at run time.

For instance, if we add this as a unit test here, we will automatically run this validation against every material in the repo:

https://github.com/terrapower/armi/blob/main/armi/materials/tests/test_materials.py#L28-L34

A benefit of that is we can control where this validation gets done. 

For instance, what if this Component is made of a void / empty material. Do we expect that material to have the `linearExpansionPercent()` method?

I think I will open a ticket to move this to unit tests.

Reply:
I'll handle it:  https://github.com/terrapower/armi/issues/861

Reply:
1. This is checking the user input, so it's not possible to perform the check as a unit test.
2. #860 removes the code in question because it fixed the root of the problem.